crates/ruma-signatures/src/keys.rs

```diff
@@ -87,23 +90,22 @@ impl Ed25519KeyPair {
     /// generated from the private key. This is a fallback and extra validation against
     /// corruption or
     pub fn from_der(document: &[u8], version: String) -> Result<Self, Error> {
-        #[cfg(feature = "ring-compat")]
-        use self::compat::CompatibleDocument;
         use pkcs8::der::Decode;
 
-        #[cfg(feature = "ring-compat")]
-        let backing: Vec<u8>;
         let oak;
 
         #[cfg(feature = "ring-compat")]
         {
+            use self::compat::CompatibleDocument;
+
             oak = match CompatibleDocument::from_bytes(document) {
                 CompatibleDocument::WellFormed(bytes) => {
                     PrivateKeyInfo::from_der(bytes).map_err(Error::DerParse)?
                 }
                 CompatibleDocument::CleanedFromRing(vec) => {
-                    backing = vec;
-                    PrivateKeyInfo::from_der(&backing).map_err(Error::DerParse)?
+                    oak = PrivateKeyInfo::from_der(&vec).map_err(Error::DerParse)?;
+
+                    return Self::from_pkcs8_oak(oak, version);
```

Comment:
Sorry for the misunderstanding, can you revert this? I think the previous version was good given your justification.

Reply:
ah gotcha :+1: